backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java

```diff
@@ -78,7 +78,33 @@ public UUID createShop(CreateShopRequest createShopRequest) {
 		}
 
 		// TODO(b/158975796): Simplify logging across all backend APIs
-		Logger.getLogger(InMemoryVirtualQueueRepository.class.getName()).log(Level.SEVERE, "Tried to fetch tokens of a shop which is not in the ACTIVE state.");
+		Logger.getLogger(InMemoryVirtualQueueRepository.class.getName()).log(Level.SEVERE,
+				"Tried to fetch tokens of a shop which is not in the ACTIVE state.");
 		return new ArrayList<>();
 	}
+
+	public Token getNewToken(UUID shopId) {
+	    if (shopMap.get(shopId).getStatus() == ShopStatus.ACTIVE) {
+              Integer newTokenNumber = shopIdToLastAllotedTokenMap.get(shopId).incrementAndGet();
+		UUID uuid = UUID.randomUUID();
+		Token newToken = new Token();
+		newToken.setTokenId(uuid);
+		newToken.setShopId(shopId);
+		newToken.setTokenNumber(newTokenNumber);
+		newToken.setStatus(Token.Status.ACTIVE);
+		tokenMap.put(uuid, newToken);
+		if (newTokenNumber != 1) {
```

Comment:
Instead of checking for inequality, check for equality here and let the else block be for != 1 case. Makes it more readable. Add a comment to depict what "1" means here.

Reply:
Done